docs/readmes/bazel/agw_with_bazel.md

```diff
@@ -0,0 +1,544 @@
+    - runtime check that all Python imports in the AGW services are bazelified.
+    - check that all C, C++ and Python files in the repository are bazelified or excluded in the `bazel/scripts/check_py_bazel.sh` script.
+- `docker-builder-devcontainer.yml`: The workflow runs on pull requests and merges to master. If necessary this workflow builds and uploads the [Docker environments](#environments). This includes in particular the Docker images that contain pre-built Bazel caches that can be used in CI or locally. These cache images are rebuild regularly.
+- `lte-integ-test-bazel-magma-deb.yml`: This workflow runs on merges to master. It runs the [LTE integration tests](#lte-integration-tests) on the AGW that have been installed via the debian package.
+- `sudo-python-tests.yml`: This workflow runs on merges to master. It runs the [Python sudo tests](#python-sudo-tests).
+
+### Bazel-diff
+
+[Tinder/bazel-diff](https://github.com/Tinder/bazel-diff) is an open-source tool that can determine which Bazel targets are impacted by the changes between two git commits. Bazel-diff is used in the `bazel.yml` [workflow](#workflows) on pull requests to determine if the changes impact
+
+- the debian package/release build
+- the Python service import statements
+- C, C++, Python and Go AGW unit test targets
+
+The granularity for the unit tests is individual Bazel test targets, which typically include a single file.
+
+For ease of use the Bazel-diff tool has been wrapped in a Bash script `bazel/scripts/bazel_diff.sh`. To determine the impacted targets from after the commit `<GIT_SHA_PRE>` up to and including the commit `<GIT_SHA_POST>`, run
+
+```bash
+cd $MAGMA_ROOT
+bazel/scripts/bazel_diff.sh <GIT_SHA_PRE> <GIT_SHA_POST>
+```
+
+The script then outputs the list of impacted targets on stdout.
+
+#### Filtering the Bazel-diff output
+
+Using [`bazel query`](#querying) the output of Bazel-diff can be filtered, e.g. for test targets or the release target. In CI this is done with the script `bazel/scripts/filter_test_targets.sh`, which takes a Bazel target rule name as a mandatory argument and a tag as an optional second argument.
+
+To filter a list of targets, contained in a file `<IMPACTED_TARGETS_FILE>`, for test targets of any language, run
+
+```bash
+bazel/scripts/filter_test_targets.sh ".*_test" < "<IMPACTED_TARGETS_FILE>"
+```
+
+The script will then print the test targets to stdout.
+
+To filter a list of targets for Python services, run
+
+```bash
+bazel/scripts/filter_test_targets.sh "py_binary" "service" < "<IMPACTED_TARGETS_FILE>"
+```
+
+To filter for specific individual targets, like the release build, use `grep`, e.g.
+
+```bash
+grep --quiet '//lte/gateway/release:release_build' "<IMPACTED_TARGETS_FILE>"
+```
+
+## Release build
+
+To build the Magma and SCTPD debian packages, with [production configuration](#configurations), locally, run
+
+```bash
+bazel run //lte/gateway/release:release_build --config=production
```

Comment:
This also builds the DHCP Helper CLI tool as a separate debian package, in case this is relevant to mention here.